sdk/textanalytics/ai-text-analytics/review/ai-text-analytics.api.md

```diff
@@ -235,6 +238,7 @@ export class TextAnalyticsClient {
     recognizeLinkedEntities(inputs: TextDocumentInput[], options?: RecognizeLinkedEntitiesOptions): Promise<RecognizeLinkedEntitiesResultCollection>;
     recognizePiiEntities(inputs: string[], language?: string, options?: RecognizePiiEntitiesOptions): Promise<RecognizeEntitiesResultCollection>;
     recognizePiiEntities(inputs: TextDocumentInput[], options?: RecognizePiiEntitiesOptions): Promise<RecognizeEntitiesResultCollection>;
+    setSubscriptionKey(subscriptionKey: string): void;
```

Comment:
I'm not sure I agree we should expose this convenience method on the client.  It's additive if we wanted to add it later.  Is there a compelling reason to do this in JS?

Reply:
Really this is about discoverability for me. I don't think we encourage anyone to hold on to a reference to a Credential after they pass it in, so it's unlikely they would look into IntelliSense on it to notice it can be updated later. 

You can kinda fish it back out of the client, but then it's a polymorphic base type instead of the concrete one since we accept both credentials and tokens.

My hypothesis is if we just tack a method on the credential and then ask someone to update their subscription key in a user study nobody will figure it out.

Reply:
I'm inclined to leave this in for discoverability, unless rotating keys isn't very important as a scenario. @bterlson  thoughts?